docs/source/tutorials_source/package/tutorial_pretrain_detectron2.py

```diff
@@ -0,0 +1,254 @@
+
+In this tutorial we show how you can do self-supervised pre-training of a
+Detectron2 backbone with lightly. The focus of this tutorial is on how to get
+and store a pre-trained ResNet50 backbone of the popular Detectron2 framework.
+If you want to learn more about self-supervised learning in general, go check
+out the following tutorials:
+
+ - :ref:`lightly-moco-tutorial-2`
+ - :ref:`lightly-simclr-tutorial-3`:
+ - :ref:`lightly-simsiam-tutorial-4`
+
+What you will learn:
+
+- How to retrieve a Detectron2 ResNet50 backbone for pre-training
+- How to do self-supervised learning with the backbone
+- How to store the backbone in a checkpoint file which can be used by Detectron2
+
+Prerequisites:
+----------------
+For the purpose of this tutorial you will need to:
+
+- Install Lightly: Follow the `instructions <https://docs.lightly.ai/getting_started/install.html>`_.
+- Install Detectron2: Follow the `instructions <https://detectron2.readthedocs.io/en/latest/tutorials/install.html>`_.
+- Download a dataset for pre-training (we will use the `SKU-110K <https://github.com/eg4000/SKU110K_CVPR19>`_ dataset).
+
+.. note::
+
+    The SKU-110K dataset is an object detection benchmark for densely packed scenes. It consists of more than 10'000
+    images showing retail shelves from varying angles and different camera models. Since contrastive learning works
+    especially well on object-centric images, cropping the objects prior to pre-training would be beneficial. However,
+    we will use the original images for simplicity. Take a look at `lightly-crop` (:ref:`lightly-command-line-tool`)
+    for more info on how to easily extract bounding box crops from your dataset.
+
+Finally, you will need the Detectron2 configuration files. They are available `here <https://github.com/facebookresearch/detectron2/tree/main/configs>`_.
+In this tutorial we use a Faster RCNN with a feature pyramid network (FPN), so make sure you have the relevant file (Base-RCNN-FPN.yaml) in your directory.
+
+
+"""
+
+# %%
+# Imports
+# -------
+#
+# Import the Python frameworks we need for this tutorial.
+import torch
+import lightly
+from detectron2 import config, modeling
+from detectron2.checkpoint import DetectionCheckpointer
+
+# %%
+# Configuration
+# -------------
+# Let's set the configuration parameters for our experiments.
+#
+# We use a batch size of 64 and an input size of 128 in order to fit everything
+# on the available amount of memory on our GPU (16GB). The number of features
+# is set to the default output size of the ResNet50 backbone.
+#
+# We only train for 5 epochs because the focus of this tutorial is on the
+# integration with Detectron2.
+
+num_workers = 8
+batch_size = 64
+input_size = 128
+num_ftrs = 2048
+lr = 0.001
+
+seed = 1
+max_epochs = 5
+
+# use cuda if possible
+device = 'cuda' if torch.cuda.is_available() else 'cpu'
+
+# some images in the dataset are truncated, use this to prevent errors
```

Comment:
This appears rather random. I would make sure all imports are at the beginning of the file. And can we elaborate more about what it means if images are truncated? 
How about using another dataset?

Reply:
We could use a different dataset, I'll check.